src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql

```diff
@@ -946,26 +953,184 @@ define ReportStoolSoftenersAndLaxatives:
+      rxNormCode: FHIRHelpers.ToCode((O.medication as FHIR.CodeableConcept).coding[0]),
+      medicationName: Calc.GetMedicationName(rxNormCode),
+      // NOTE: Assuming a single dosage instruction element
+      dosageInstruction: O.dosageInstruction[0],
+      // NOTE: Assuming a single dose and rate element
+      doseAndRate: dosageInstruction.doseAndRate[0],
+      repeat: dosageInstruction.timing.repeat,
+      frequency: Coalesce(repeat.frequencyMax.value, repeat.frequency.value),
+      period: System.Quantity { value: repeat.period.value, unit: repeat.periodUnit.value },
+      doseDescription:
+        Coalesce(
+          // There should be a conversion from FHIR.SimpleQuantity to System.Quantity
+          if doseAndRate.dose is FHIR.Range
+            then ToString((doseAndRate.dose as FHIR.Range).low)
+                + '-' + ToString((doseAndRate.dose as FHIR.Range).high)
+                + (doseAndRate.dose as FHIR.Range).high.unit.value
+            else ToString(FHIRHelpers.ToQuantity(doseAndRate.dose as FHIR.SimpleQuantity)),
+            ''
+        ),
+      frequencyDescription:
+        ToString(dosageInstruction.timing.repeat.frequency.value) +
+          Coalesce(
+            '-' + ToString(dosageInstruction.timing.repeat.frequencyMax.value),
+            ''
+          )
+    return {
+      rxNormCode: rxNormCode,
+      isDraft: O.status.value = 'draft',
+      // NOTE: Assuming asNeeded is expressed as a boolean
+      isPRN: dosageInstruction.asNeeded,
+      prescription:
+        if dosageInstruction.text is not null then
+          medicationName + ' ' + dosageInstruction.text.value
+        else
+          // TODO: Shouldn't need the .value here on asNeededBoolean
+          medicationName + ' ' + doseDescription + ' q' + frequencyDescription + (if dosageInstruction.asNeeded then ' PRN' else ''),
+      dose: if doseAndRate.dose is FHIR.Range
+            then (doseAndRate.dose as FHIR.Range).high
+            else FHIRHelpers.ToQuantity(doseAndRate.dose as FHIR.SimpleQuantity),
+      dosesPerDay: Coalesce(Calc.ToDaily(frequency, period), 1.0)
+    }
+
+//define TestPrescriptions:
+//  Prescriptions({ { rxNormCode: Code '313993' from RxNorm, doseQuantity: Quantity { value: 14, unit: 'mg' }, dosesPerDay: 2.00 } })
+
+define function MME(prescriptions List<MedicationRequest>):
+  (Prescriptions(prescriptions)) P
+    let mmePerIngredient: Calc.CalculateMMEs({ { rxNormCode: P.rxNormCode, doseQuantity: P.dose, dosesPerDay: P.dosesPerDay } })
+    return {
+      rxNormCode: P.rxNormCode,
+      isDraft: P.isDraft,
+      isPRN: P.isPRN,
+      prescription: P.prescription,
+      dailyDose: Combine(mmePerIngredient X return X.dailyDoseDescription, '\r\n'),
+      mme: Sum(mmePerIngredient X return X.mme)
+    }
 
-// MME (MOST RECENT w/ lookback 6 months) - amount and date
+define function TotalMME(prescriptions List<MedicationRequest>):
```

Comment:
@mldorr Maggie, I am not sure what this function is doing..is it calculating the MME for **all** the prescriptions or MME for a particular prescription on different dates perhaps?